src/runtime/io.cpp

```diff
@@ -828,6 +828,36 @@ extern "C" LEAN_EXPORT obj_res lean_io_rename(b_obj_arg from, b_obj_arg to, lean
     return io_result_mk_ok(box(0));
 }
 
+/* createTempFile : IO (Handle × FilePath) */
+extern "C" LEAN_EXPORT obj_res lean_io_create_tempfile(lean_object * /* w */) {
+    char path[PATH_MAX];
+#if defined(LEAN_WINDOWS)
+    // https://learn.microsoft.com/en-us/windows/win32/fileio/creating-and-using-a-temporary-file
+    DWORD retval = GetTempPath(MAX_PATH, path);
+    if (retval > MAX_PATH || (retval == 0)) {
+        return io_result_mk_error((sstream() << GetLastError()).str());
+    }
+    // If the temp dir is so long that we can't put files into it something is seriously wrong.
+    const char* file_pattern = "tmp.XXXXXXXX";
+    const int file_pattern_size = strlen(file_pattern);
+    lean_always_assert(PATH_MAX >= strlen(path) + strnlen(file_pattern, file_pattern_size) + 1);
+    strncat(path, file_pattern, file_pattern_size);
+#else
+    const char* path_template = "/tmp/tmp.XXXXXXXX";
```

Comment:
No love for `TMPDIR`? Is that what Rust does?

Reply:
It would appear they do respect `TMPDIR` and default to `/tmp` otherwise, I'll change that.